Document reprovision param for Update Workflow API (#8172)
opensearch-project/documentation-website: _automating-configurations/api/create-workflow.md

You cannot specify both the `provision` and `update_fields` parameters at the same time.
{: .note}

If a workflow has been provisioned, you can update and reprovision the full template:

```json
PUT /_plugins/_flow_framework/workflow/<workflow_id>?reprovision=true
{
  <updated complete template>
}
```

> You can add new steps to the workflow but not delete them. Only steps for index settings, search pipelines, and ingest pipelines are currently supported for updating.

Reviewer: ```suggestion
You can add new steps to the workflow but cannot delete them. Only index setting, search pipeline, and ingest pipeline steps can currently be updated.
```

You cannot specify both the `provision` and `update_fields` parameters at the same time.
{: .note}

> If a workflow has been provisioned, you can update and reprovision the full template:

Author: I expect this section to create a merge conflict when #8171 is merged. This section should go above that new content as it's directly related to updating.  I'll resolve conflicts when the time comes.

```json
PUT /_plugins/_flow_framework/workflow/<workflow_id>?reprovision=true
{
  <updated complete template>
}
```

> {% include copy-curl.html %}

Reviewer 2: can be removed I assume

Author: I'll leave that to doc reviewers.  I see the `<placeholder>` stuff used in other copy/paste.

Reviewer 2: Yeah I see that now in other locations

Reviewer 3: We typically want to spell out everything in the documentation in order to avoid confusion. This request clearly shows the `reprovision` query parameter and specifies that it needs to be a _complete_ template, so we should keep this here. The template is very big so it's fine to use a descriptive placeholder.

You can add new steps to the workflow but not delete them. Only steps for index settings, search pipelines, and ingest pipelines are currently supported for updating.

> {: .note}

Reviewer 2: The {: .note} is not producing anything specific in markdown, can we remove?

Author: I think that renders on the actual documentation page, see the similar one above which renders like this:

![Screenshot 2024-09-04 at 4 51 44 PM](https://github.com/user-attachments/assets/a896df54-7b35-4b77-ba7d-10e5e9fbeeac)

Reviewer 2: I see, I am okay with it then

Author: I anticipate we'll make this particular paragraph more detailed and link to a whole separate page (like the use case templates) for the remainder of the feature, so just trying to add the minimum this time.
